rfcs/0001-reason-message-diff.md

```diff
@@ -2,52 +2,291 @@
+- Indentation should be preserved whenever possible in the diffs 
+
+See mockup images attached in comment to thread 
+
+------------------------------
+
+**For Reason "TRIGGER"**
 
-Reason: [Reason-Code] - [Reason-Message]
+- Print the name of the user and the time () it was triggered as detail for the reason
+- No diff
+
+------------------------------
+
+$kp build status petclinic
+Image:      index.docker.io/username/instance@sha256:fsdgh39478th0g9tuisodfhgvns938e54iunwuiehfasf9wgiauwsfx
+Status:     SUCCESS
+Reason:     TRIGGER - A new build was manually triggered by [user] at [time].
+
+Builder:    our-special-url.jfrog.io/instance/cluster-builder:default@sha4398ths9rdpfp9sghbne985-s9ed9hgbd9rz4sh98dxbfs
+Run Image:  index.docker.io/paketobuildpacks/run@sha256:34978twgsui4789et0wh9384erhsg0e798riuhgs0er7iugshe98riugeh99rgaser89
+
+Source:   Local Source 
+
+BUILDPACK ID                            BUILDPACK VERSION
+paketo-buildpacks/bellsoft-liberica     2.13.0
+paketo-buildpacks/maven                 2.3.0
+paketo-buildpacks/executable-jar        2.1.1
+paketo-buildpacks/apache-tomcat         1.5.0
+paketo-buildpacks/dist-zip              1.4.0
+paketo-buildpacks/spring-boot           2.5.0
+
+------------------------------
 
-Examples: 
 **For Reason "COMMIT"**
-- Ex. Reason: COMMIT - Received a new commit with ID [truncated-SH] from branch [branch-name]. 
+
+- Print a diff below the reason section that shows the change between the two revisions or the two branch names, whichever changed. 
+
+------------------------------
+
+$kp build status petclinic
+Image:      index.docker.io/username/instance@sha256:fsdgh39478th0g9tuisodfhgvns938e54iunwuiehfasf9wgiauwsfx
+Status:     SUCCESS
+Reasons:    COMMIT 
+            - Revision: 234890hpe9vjsr8gse98rghser
+            + Revision: 43t789wghges87h540eq8378ge
+
+Builder:    our-special-url.jfrog.io/instance/cluster-builder:default@sha4398ths9rdpfp9sghbne985-s9ed9hgbd9rz4sh98dxbfs
+Run Image:  index.docker.io/paketobuildpacks/run@sha256:34978twgsui4789et0wh9384erhsg0e798riuhgs0er7iugshe98riugeh99rgaser89
+
+Source:   Local Source 
+
+BUILDPACK ID                            BUILDPACK VERSION
+paketo-buildpacks/bellsoft-liberica     2.13.0
+paketo-buildpacks/maven                 2.3.0
+paketo-buildpacks/executable-jar        2.1.1
+paketo-buildpacks/apache-tomcat         1.5.0
+paketo-buildpacks/dist-zip              1.4.0
+paketo-buildpacks/spring-boot           2.5.0
+
+------------------------------
 
 **For Reason "CONFIG"**
-- Ex. Reason: CONFIG - The [image-name] image config was updated. See diff below.
-[output of diffing library on the two image config files]
 
-**For Reason "BUILDPACK"**
-- Ex. Reason: BUILDER - The buildpacks used by your app have changed. See diff below. 
-[output of diffing library on the two Bill of Materials' for the builds]
+- Print a diff beneath the reason section
+- Print the three lines preceding where the diff begins in order to show context for the change. 
```

Comment:
Could it just be simpler to print the config for build A with the `-` symbol next to the fields that have changed and then print the config for build B with the `+` symbol next to the fields that are different from build A?

Reply:
I agree I think it should be the whole config because if multiple config values change it would be hard to satisfy this constraint:

> Print the three lines preceding where the diff begins in order to show context for the change. 

I think we should just print the whole config but I think it should have a line-by-line diff so that changes are clear, not both configs listed entirely:

```
  build:
    env:
    - name: BP_JAVA_VERSION
-     value: 11.*
+     value: 14.*
- resources: {}
+ resources:
+   limits:
+     cpu: 1M
- CacheSize: 1M
+ CacheSize: 12M
  unchanged: value
  rest:
    of:
      config: foo
```